main.cpp

```diff
@@ -21,41 +24,72 @@ cstring remove_extension(cstring filename) {
     return filename.substr(0, idx);
 }
 
+const IR::P4Program *prune_statements(const IR::P4Program *program,
+                                      P4PRUNER::PrunerOptions options,
+                                      int required_exit_code) {
+    cstring stripped_name = remove_extension(options.file);
+    stripped_name += "_stripped.p4";
```

Comment:
Instead of always producing a "temp.p4" file we instead append to the filename. This also allows for multiple executions on different files in parallel.